sdk/core/AzureCore/Source/DataStructures/XMLModel.swift

```diff
@@ -198,6 +198,7 @@ internal class XMLTreeNode {
     var dictionary: [String: Any]? {
         var propDict = [String: Any]()
         for (key, metadata) in properties {
+            assert(metadata.parent == nil, "Unexpectedly found a parent reference, which could leak to memory leaks.")
```

Comment:
The Swift compiler will strip this out for release builds, but I consider it a good safeguard against regression, especially given that only running the demo in profiling mode caught this leak.